src/dbSta/test/hierwrite.vok

```diff
@@ -0,0 +1,2691 @@
+module gcd (clk,
+    req_rdy,
```

Comment:
There doesn't seem to be any correct wire declarations in this verilog file and the wire names prevent seem to be using flat names instead of the correct hierarchical names

Reply:
Yes, it is intentional that the flat names are used for the wires and they are not declared.

Reply:
Then this is not a correct verilog file (ie. It's not equivalent to the input anymore). This seems wrong to me

Reply:
By "not equivalent" I think you are trying to say "not syntactically of the same style".  Or are you able to show they are not logically equivalent ? (which would be interesting and useful).

Verilog allows implicit wires, and if you look carefully you will see that the nets connected to the ports of the modules are what you would probably regard as correct (ie not involving hierarchical names) and it is only the internal nets which have their flat names. And yes, those are the flat net names on the database and it is an intentional choice, for now.

I have intentionally kept the internal net names flat  so I can debug the backend tools with hierarchy (I want to see things like buffer insertion having the correct flat name in the correct scope). As you might imagine it is not very difficult to get the hierarchical name by stripping of the name prefix within the scope.

Once we are done with bringing up the backend tools then the syntax will be cleaned up to a pretty form. 

Peter, please don't block this very minor pull request which is useful for us to be able to debug some of the backend stuff in the flow (you might imagine that it is useful to be able to write out the verilog after each stage to see the perturbation of the design). The hierarchy stuff is hidden behind flags and there are only a few test cases which exercise it. Beautifying the Verilog output is not too hard and can be usefully done once we have cts, resizer and the rest working with hierarachy. Thanks.

Reply:
The net names should be the same, if this is actually to be useful between levels, if the nets are changing with this each time this verilog is used the net names will just get longer as the hierarchical names keep getting prepended. If it's an easy fix to strip the prefixes then I would say that should be done before merging.

I have not blocking this from merging, but I don't think it should be merged while the netnames are not correctly honoring the hierarchical names (as you said it was not difficult to fix so we should do that). If you need a debugging mode where you don't strip the names that is fine for you to add, but we shouldn't be exposing the debugging information to users.

Reply:
congratulations, you just blocked the merge and slowed progress. Again, I intentionally put in the flat names to catch the latent bugs in the backend (for example look at the net name creation routine makeUniqueNetName in Resizer.cc which creates the wrong name in the  wrong scope. To catch that we needed to write out the flat names). Once that is fixed then sure we can put in a "pretty print verilog fix". There will doubtless be other things than net names that people would want printed in the verilog output.

Reply:
We can have it merged now, but please create an issue regarding the future fix regarding the verilog net names.